cmd/peek-submission/main_test.go

```diff
@@ -1,41 +1,88 @@
 package main
 
 import (
+	"fmt"
 	"testing"
 
-	"github.com/edulinq/autograder/internal/api/core"
-	"github.com/edulinq/autograder/internal/api/courses/assignments/submissions"
 	"github.com/edulinq/autograder/internal/cmd"
-	"github.com/edulinq/autograder/internal/util"
 )
 
+var testCases = []struct {
+	cmd.CommonCMDTestCases
+	targetEmail      string
+	courseID         string
+	assignmentID     string
+	targetSubmission string
+}{
+	{
+		CommonCMDTestCases: cmd.CommonCMDTestCases{
+			ExpectedStdout: LATEST_SUBMISSION,
+		},
+		targetEmail:  "course-student@test.edulinq.org",
+		courseID:     "course101",
+		assignmentID: "hw0",
+	},
+	{
+		CommonCMDTestCases: cmd.CommonCMDTestCases{
+			ExpectedStdout: SPECIFIC_SUBMISSION_SHORT,
+		},
+		targetEmail:      "course-student@test.edulinq.org",
+		courseID:         "course101",
+		assignmentID:     "hw0",
+		targetSubmission: "1697406272",
+	},
+	{
+		CommonCMDTestCases: cmd.CommonCMDTestCases{
+			ExpectedStdout: SPECIFIC_SUBMISSION_LONG,
+		},
+		targetEmail:      "course-student@test.edulinq.org",
+		courseID:         "course101",
+		assignmentID:     "hw0",
+		targetSubmission: "course101::hw0::student@test.com::1697406256",
+	},
+	{
+		CommonCMDTestCases: cmd.CommonCMDTestCases{
+			ExpectedStdout: NO_SUBMISSION,
+		},
+		targetEmail:  "course-admin@test.edulinq.org",
+		courseID:     "course101",
+		assignmentID: "hw0",
+	},
+	{
+		CommonCMDTestCases: cmd.CommonCMDTestCases{
+			ExpectedStdout: INCORRECT_SUBMISSION,
+		},
+		targetEmail:      "course-student@test.edulinq.org",
+		courseID:         "course101",
+		assignmentID:     "hw0",
+		targetSubmission: "ZZZ",
+	},
+	{
+		CommonCMDTestCases: cmd.CommonCMDTestCases{
+			ExpectedExitCode: 2,
+			ExpectedStderr:   `{"message":"Could not find course: 'ZZZ'."}`,
```

Comment:
Just query for the string "Could not find course: 'ZZZ'.".
Including the rest makes it harder for people to see what the important part is.